### ISSUE TITLE
Same-line comments break at-rule-empty-line-before

### ISSUE THREAD

**Issue by timkelty** [stylelint/stylelint#2583]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

With the config below, having a comment on the same line as the at-rule invalidate at-rule-empty-line-before when it should not.

> Which rule, if any, is this issue related to?

`at-rule-empty-line-before` is where I noticed it, but possibly an issue with others.

> What CSS is needed to reproduce this issue?

```css
@custom-media --xsm-viewport (width < 576px); /* portrait phones */
@custom-media --sm-viewport (width >= 576px) and (width < 768px); /* landscape phones */
```

> What stylelint configuration is needed to reproduce this issue?

```json
  rules: {
    "at-rule-empty-line-before": [
      "always", {
        except: ["blockless-after-blockless"],
      }
    ],
  }
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
2:1	⚠  Expected empty line before at-rule (at-rule-empty-line-before) [stylelint]
```

**Comment by jeddy3** [comment]
@timkelty Thanks for the report and for using the template.

This has been fixed and will be released in [`8.0.0`](https://github.com/stylelint/stylelint/blob/v8/CHANGELOG.md#800). There's [quite a few issues left to go in the milestone](https://github.com/stylelint/stylelint/milestone/23), so please consider helping out if you'd like to see the release drop sooner.

**Comment by timkelty** [comment]
@jeddy3 thanks! I browsed through the roadmaps first but missed it somehow.
